docs/proposals/authentication-filter.md

```diff
@@ -22,3 +22,1146 @@ This new filter should eventually expose all forms of authentication available t
+
+// AuthenticationFilter configures request authentication (Basic or JWT) and is
+// referenced by HTTPRoute filters via ExtensionRef.
+type AuthenticationFilter struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  // Spec defines the desired state of the AuthenticationFilter.
+  Spec AuthenticationFilterSpec `json:"spec"`
+
+  // Status defines the state of the AuthenticationFilter, following the same
+  // pattern as SnippetsFilter: per-controller conditions with an Accepted condition.
+  //
+  // +optional
+  Status AuthenticationFilterStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// AuthenticationFilterList contains a list of AuthenticationFilter.
+type AuthenticationFilterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []AuthenticationFilter `json:"items"`
+}
+
+// AuthenticationFilterSpec defines the desired configuration.
+// Exactly one of Basic or JWT must be set according to Type.
+// +kubebuilder:validation:XValidation:message="for type=Basic, spec.basic must be set and spec.jwt must be empty; for type=JWT, spec.jwt must be set and spec.basic must be empty",rule="self.type == 'Basic' ? self.basic != null && self.jwt == null : self.type == 'JWT' ? self.jwt != null && self.basic == null : false"
+// +kubebuilder:validation:XValidation:message="type 'Basic' requires spec.basic to be set. All other spec types must be unset",rule="self.type == 'Basic' ? self.type != null && self.jwt == null : true"
+// +kubebuilder:validation:XValidation:message="type 'JWT' requires spec.jwt to be set. All other spec types must be unset",rule="self.type == 'JWT' ? self.type != null && self.basic == null : true"
+// +kubebuilder:validation:XValidation:message="when spec.basic is set, type must be 'Basic'",rule="self.basic != null ? self.type == 'Basic' : true"
+// +kubebuilder:validation:XValidation:message="when spec.jwt is set, type must be 'JWT'",rule="self.jwt != null ? self.type == 'JWT' : true"
+type AuthenticationFilterSpec struct {
+  // Type selects the authentication mechanism.
+  Type AuthType `json:"type"`
+
+  // Basic configures HTTP Basic Authentication.
+  // Required when Type == Basic.
+  //
+  // +optional
+  Basic *BasicAuth `json:"basic,omitempty"`
+
+  // JWT configures JSON Web Token authentication (NGINX Plus).
+  // Required when Type == JWT.
+  //
+  // +optional
+  JWT *JWTAuth `json:"jwt,omitempty"`
+}
+
+// AuthType defines the authentication mechanism.
+// +kubebuilder:validation:Enum=Basic;JWT
+type AuthType string
+
+const (
+  AuthTypeBasic AuthType = "Basic"
+  AuthTypeJWT   AuthType = "JWT"
+)
+
+// BasicAuth configures HTTP Basic Authentication.
+type BasicAuth struct {
+  // SecretRef allows referencing a Secret in the same namespace
+  SecretRef LocalObjectReferenceWithKey `json:"secretRef,omitempty"`
```

Comment:
If we're never going to support ConfigMaps for this (do we really need to?), then we probably don't actually need to use a LocalObjectReference that specifies the Kind. We could support just Name, Namespace, Key. But we should only do this if we know for sure, because otherwise it'll make our API a little messier in the future.

Reply:
I think we should just keep it as `SecretRef`.
If gives us the flexibility of eventually supporting ConfigMap, even if we never choose to.
It also has the added benefit of being familiar to users and it's very clear what it's trying to accomplish.

Reply:
@sjberman After going back and forth, it feels like it would be more secure to stick with just secret ref. The main benefit that ConfigMap offers is less overhead in encoding and decoding and it would be faster when dealing with less sensitive data data. As I looked over it, we mainly deal with usernames/passwords and JWKs in these scenarios, which I feel should always be encoded.